internal/nginx/config/servers.go

```diff
@@ -198,6 +198,14 @@ func createReturnValForRedirectFilter(filter *v1beta1.HTTPRequestRedirectFilter,
 	port := listenerPort
 	if filter.Port != nil {
 		port = int32(*filter.Port)
+	} else {
+		if filter.Scheme != nil {
```

Comment:
this contradicts the statement here about well-known ports -- https://github.com/kubernetes-sigs/gateway-api/blob/a5ede12430c31639349ae6c00666890591f72a40/apis/v1beta1/httproute_types.go#L928-L934 🤔

Reply:
note: this way, we will not need if filter.Port == nil { at all

> Well, we will still need to cover the case where filter.Port == nil && scheme is set.

perhaps like this?

```
        port := listenerPort
	if filter.Port != nil {
		port = int32(*filter.Port)
	}

	hostnamePort := fmt.Sprintf("%s:%d", hostname, port)

	scheme := "$scheme"
	if filter.Scheme != nil {
		scheme = *filter.Scheme
	}
	
	// Don't specify the port in the return url if the scheme is
        // well known and the port is already set to the correct well known port
        if (port == 80 && scheme == "http") || (port == 443 && scheme == "https") {
	  hostnamePort = hostname
     }
```